Assets/Scripts/UIManager.cs

```diff
@@ -0,0 +1,321 @@
+
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+
+public class UIManager : MonoBehaviour
+{
+    [SerializeField] private TextMeshProUGUI summonName1;
+    [SerializeField] private TextMeshProUGUI summonName2;
+    [SerializeField] private TextMeshProUGUI summonName3;
+    [SerializeField] private TextMeshProUGUI summonName4;
+    [SerializeField] private TextMeshProUGUI summonName5;
+
+    [SerializeField] private Image summonImage1;
+    [SerializeField] private Image summonImage2;
+    [SerializeField] private Image summonImage3;
+    [SerializeField] private Image summonImage4;
+    [SerializeField] private Image summonImage5;
+
+    [SerializeField] private Sprite ghostImage;
+    [SerializeField] private Sprite skeletonImage;
+    [SerializeField] private Sprite zombieImage;
+
+    [SerializeField] private GameObject summonIndicator1;
+    [SerializeField] private GameObject summonIndicator2;
+    [SerializeField] private GameObject summonIndicator3;
+    [SerializeField] private GameObject summonIndicator4;
+    [SerializeField] private GameObject summonIndicator5;
+
+    private int i = 0;
+
+    public static UIManager Instance
+    {
+        get;
+        private set;
+    }
+
+    private void Awake()
+    {
+        if (Instance != null && Instance != this)
+        {
+            Destroy(this);
+        }
+        else
+        {
+            Instance = this;
+        }
+    }
+
+    public void UpdateSummon(string summon, int summonCount)
```

Comment:
Pass a list of selected summons instead of the name and count, inside instantiate the new ui prefab and setup the sprite and name(use selectedSummon.GetType().ToString()))

Reply:
Done